### ISSUE TITLE
A lot of unused closure callback

### ISSUE THREAD

**Issue by peepo3663** [ananthbh/SurveyList#6]
I saw a lot of callbacks which is not used or maintained at all for eg.
```swift
checkAuthToken { }
```

Callback inside `checkAuthToken` If you didn't plan to use it mark it as optional so you will not have another side effects here
and also Inside this method you didn't call callback every cases
for eg.
```swift
case .failure:
                print("failed in fetching data")
            }
```
you just printed it out but not used callback so this could cause deadlock in your project

**Comment by ananthbh** [comment]
yeah it's true that I did not handle them. regardign the `checkAuthToken` it's more to do some updating if needed after the authToken is checked. but it's really needed in all the cases and I agree.
